pkg/validate/dynamic/dynamic.go

```diff
@@ -764,23 +764,41 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			)
 		}
 
-		_, net, err := net.ParseCIDR(*ss.AddressPrefix)
-		if err != nil {
-			return err
+		// Handle both addressPrefix & addressPrefixes
+		if ss.AddressPrefix == nil {
+			for _, address := range *ss.AddressPrefixes {
+				if err = validateSubnetSize(s, address); err != nil {
+					return err
+				}
```

Comment:
If this is validating a range, do we want to error if the first element of the slice doesn't validate?

Reply:
Honestly I'm not sure. 

I'm not even sure how openshift will handle this.  You can technically specify multiple IP address ranges on a subnet now (wasn't that the point of a subnet to begin with!??). 

For now, we've only seen customers with a single one defined.  You have to opt into the MSFT AFEC feature flag `Microsoft.Network/AllowMultipleAddressPrefixesOnSubnet` in order to use it for now.

Reply:
>I'm not even sure how openshift will handle this. You can technically specify multiple IP address ranges on a subnet now (wasn't that the point of a subnet to begin with!??).
>
>For now, we've only seen customers with a single one defined. You have to opt into the MSFT AFEC feature flag Microsoft.Network/AllowMultipleAddressPrefixesOnSubnet in order to use it for now.

I poked around in the installer code and nothing I found indicated that the installer will be unhappy if you provide a subnet with multiple address ranges. Complicated stuff though, so it would be good to ask the installer folks to verify.

If OCP doesn't support multiple address ranges, we can error out here if the cx has put more than one and then the original question in this thread is moot.

Reply:
Thanks for the investigation.  

I also poked around in machine-config-provider-azure and there is never an explicit subnet address assignment used during NIC provisioning, so it should be handled transparently.  The only thing that is specified during creation is the subnet ID.  

So I think the proper validation here would be to add up all the address spaces and make sure the total address space is >27 mask, also taking into account the number of reserved IP addresses in smaller address spaces may have an impact.

Reply:
It's not perfect, but instead of adding up individual ranges, I'm going to validate that at least one address prefix has a /27 or greater and error out if not.  

If we have a bunch of /30s it's not going to make sense as there isn't enough address space for anything there.  The cases around this are going to get crazy, and at least we're returning a valid message to the customer.